DOCSP-13862 delete one usage example (#6)
mongodb/docs-golang: source/usage-examples/deleteOne.txt

=================

.. default-domain:: mongodb

You can delete a single document in a collection using the
``DeleteOne()`` method.

The following example specifies a query filter that matches documents in
the ``movies`` collection with the value "The Room" in the ``title`` field, and
deletes the first document that matches. 

.. include:: /includes/quick-start/run-example-tip.rst

.. literalinclude:: /includes/quick-start/deleteOne.go
   :start-after: begin deleteOne
   :end-before: end deleteOne
   :emphasize-lines: 3
   :language: go
   :dedent:

> Click here <TODO> to see a fully runnable example.

Author: The findOne and updateOne currently have placeholders -- I believe this is going to be one of the clean up tickets to properly link after all tickets are done

=================

.. default-domain:: mongodb

You can delete a single document in a collection using the
``DeleteOne()`` method.

The following example specifies a query filter that matches documents in
the ``movies`` collection with the value "Twilight" in the ``title`` field, and
deletes the first document that matches. 

.. include:: /includes/quick-start/run-example-tip.rst

.. literalinclude:: /includes/quick-start/deleteOne.go
   :start-after: begin deleteOne
   :end-before: end deleteOne
   :emphasize-lines: 2
   :language: go
   :dedent:

Click here <TODO> to see a fully runnable example.

Expected Result
---------------

> After deleting this document, you should not be able to find it in the

Reviewer: ```suggestion
After running the above code snippet, you should not be able to find the following document in the
```
Suggestion: I think "after deleting... you shouldn't be able to find" is a tautological statement. I think "running the above code snippet" better conveys the task the just did (which may or may not have actually deleted the document). 

Suggestion: I think "this document" should be "the following" document

the following document in the ``movies`` collection:

.. code-block:: json

    { "_id": { "$oid": "573a13bff29313caabd5e06b" }, 
      ...
      "title": "Twilight",
      ...
    }

For an example on how to find a document, see our :doc:`Find
One Usage Example </usage-examples/findOne>`.

Additional Information
----------------------

For more information on deleting documents, specifying query filters,
and handling potential errors, see our guide on <TODO:
Deleting a Document>.

API Documentation
~~~~~~~~~~~~~~~~~

> :go-api:`DeleteOne() <mongo#Collection.DeleteOne>`

Reviewer 2: Similar comment as on https://github.com/mongodb/docs-golang/pull/5:
This link in on the staging server goes to [pkg.go.dev](http://pkg.go.dev) documentation for the `github.com/mongodb/mongo-go-driver` package, but the canonical package is `go.mongodb.org/mongo-driver`. They're technically the same, but it could be confusing if the reader doesn't realize that.

Please update to link to the docs at https://pkg.go.dev/go.mongodb.org/mongo-driver instead.

=================

.. default-domain:: mongodb

You can delete a single document in a collection using the
``DeleteOne()`` method.

> The following example specifies a query filter that matches documents in
> the ``movies`` collection with the value "Twilight" in the ``title`` field, and
> deletes the first document that matches. 

Reviewer: ```suggestion
The following example deletes a document in
the ``movies`` collection of the ``sample_mflix`` database with the value "Twilight" in the ``title`` field.
```

Suggestion: I think "specify a query filter that matches" can be simplified to "deletes". I think the more detailed explanation should be included in the delete fundamentals page. 

Suggestion: I think if the collection is mentioned it makes sense to also mention the database.

Author: I didn't implement this change because it strays from the template we agreed on as a team, but I'm open to having a team discussion on how we should word the example statement.

Reviewer: Zoom call. Determined that we have different interpretations of the usage example template. Should bring this up in a team discussion later on.

.. include:: /includes/quick-start/run-example-tip.rst

.. literalinclude:: /includes/quick-start/deleteOne.go
   :start-after: begin deleteOne
   :end-before: end deleteOne
   :emphasize-lines: 2
   :language: go
   :dedent:

Click here <TODO> to see a fully runnable example.

Expected Result
---------------

You should not be able to find the following document: 

.. code-block:: json

    { "_id": { "$oid": "573a13bff29313caabd5e06b" }, 
      ...
      "title": "Twilight",
      ...
    }

> To check if you successfully deleted this document, try performing a
> find operation with the ``title`` field on the collection. For an
> example of how to perform the find operation, see our <TODO: Find One
> Usage Example guide>. 

Reviewer: ```suggestion
For an example of finding a document, see our <TODO: Find One
Usage Example>. 
```
Suggestion: I think this can be shortened.

Suggestion: I think the clause 

> "try performing a find operation with the ``title`` field on the collection"

is imprecise, and should either be deleted or made more precise.

Author: I made this more percise.

Reviewer: Zoom call. Agree the new wording should replace the intro sentence for the result.
